### ISSUE TITLE
Make random_state recoverable when passed into a reducer/clusterer model

### ISSUE THREAD

**Issue by patcon** [polis-community/red-dwarf#67]
Right now, when we don't set `random_state` arg on `run_pipeline` (this new function rename will be merged soon), it passs in the default of `None`, which means no seed is used, and the state is therefore truly random (not seeded), and there is no way to reproduce.

It would be nicer if it instead randomly generated a seed, and then passed that into any reducer/clusterer models. That way, we could retrieve the seed more easily from the model, and be able to reproduce whatever was generated, just by using the the existing `get_params()` function of the sklearn estimators that we use internally:

https://scikit-learn.org/stable/modules/generated/sklearn.base.BaseEstimator.html#sklearn.base.BaseEstimator.get_params

What I propose we'd want

```py
from reddwarf.implementations.base import run_pipeline

result = run_pipeline(random_state=None, **other_kwargs)

reducer_model = result.reducer
reducer_params = reducer_model.get_params()

# Currently returns `None`, but we want a specific value
prior_random_state = reducer_params["random_state"]

reproduced_result = run_pipeline(random_state=prior_random_state, **other_kwargs)
```
